File: src/main/php/Easy/ReformatDate.php
<?php

declare(strict_types=1);

namespace Src\Easy;

class ReformatDate
{
    public function solve(string $date): string
    {

        $months = ['Jan', 'Feb', 'Mar', 'Apr', 'May', 'Jun', 'Jul', 'Aug', 'Sep', 'Oct', 'Nov', 'Dec'];
        $parts = explode(' ', $date);
        $month = array_search($parts[1], $months) + 1;
        $day = 0;
        foreach (str_split($parts[0]) as $char) {
            if ($char >= '0' && $char <= '9') {
                $day = $day * 10 + ($char - '0');
            }
        }

        return sprintf('%s-%02d-%02d', $parts[2], $month, $day);
    }
}

File: src/easy/reformat_date.rs
pub fn reformat_date(date: String) -> String {
    let parts: Vec<&str> = date.split(" ").collect();
    let month = vec![
        "Jan", "Feb", "Mar", "Apr", "May", "Jun", "Jul", "Aug", "Sep", "Oct", "Nov", "Dec",
    ]
    .iter()
    .position(|&x| x == parts[1])
    .unwrap_or(0)
        + 1;
    let mut day: u32 = 0;
    for i in parts[0].chars() {
        if i.is_numeric() {
            day = day * 10 + i.to_digit(10).unwrap_or(0);
        }
    }
    format!("{}-{:0>2}-{:0>2}", parts[2], month, day)
}
